### PR TITLE
Create deploy.bat

### PR DESCRIPTION
**Pull request by siteleas**
This is a batch file to quick launch projects under windows

### CODE REVIEW

**Comment by Alroniks** [comment]
Thank you @siteleas 
But I think it can't be merged because your script works only for your environment and other people can use other environment configuration with other paths and other settings, so gitify as the project should be universal for all of them.

**Comment by siteleas** [comment]
@Alroniks Not really sure what you mean. This batch file receives a parameter from stdin that it uses to declare the destination folder and the base deployment folder can customized for each user environment using the gitifyModxRoot environment variable that is toward the top of the batch file. Other than that the batch file is environment agnostic. So, as long as the per-requisites (dependencies) needed by Gitify are met the batch file should run without issue (and not merely on my computer or with my setup). In fact, most of the commands in the batch file were merely copied and pasted directly from Gitify documentation. So unless you are meaning that Gitify documented commands are not system agnostic somehow, I am really not following your reasoning.

File: deploy.bat [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+REM called scripts. I then updated my Windows PATH environment variable and added the
+REM scripts folder in order to be able to run the script anywhere.
+REM
+REM Once the script has completed it's execution, it dump you into the development
+REM folder so you can continue to work on the newly created MODx build environment.
+REM
+REM Change the variable below in order to align with your configuration
+REM
+REM The root foler to create MODx install sub-folders in
+set gitifyModxRoot=k:\ampps\www
```
Comment (Mark-H) [inline deploy.bat]: I'm not familiar with writing bat scripts, but this looks like a hardcoded folder? I think that's also what @Alroniks meant ;)

**Comment by siteleas** [comment]
Hopefully the updated documentation at the top of the file clarifies things. Also, I removed the hard coded path and inserted placeholder text with documentation indicating that the end user needs to change this setting. 

You also may want to know that I am working on idiot proofing the script with error messages and prompts. 

I wanted you to get a sense for where I am going with this batch script so that I can know whether I should be committing the time to improving the script for the Gitify community or if doing so would not be a productive use of my time due to not being integrated into the modmore repository.

Please advise

**Comment by Alroniks** [comment]
When I said about this proposal I meant that I and I think Mark too, we want keep Gitify as much more clear and simple tool for their tasks without a lot of settings and tweaks for each case in the world. I agree, many people use Windows for development and it is a fact (vagrant not solves all) and Gitify should have ability to work in Windows but... In your first version of code you check your solutions in only your environment with ampps server (not sure), but also exists Denwer, a lot of MAMP/WAMP-like tools, GitBash as Mark noticed etc. And also exists https://github.com/modmore/Gitify/blob/master/Gitify.cmd for launch gitify in Windows by the proper way. 

I use mac and linux last 5 years and don't know a lot of about current state of Windows and maybe you are totally right, but we want to know why this solution should be applied. So please let's continue and please be patient when we are doubting.

**Comment by sebastian-marinescu** [comment]
The thought is interesting as I'm also using bat/sh files to get work done faster. 
But this script is for the initial setup of Gitify. That's something I don't think can be _generalized_ across different environments as every developer wants to put his stuff in a different folder. Also in my opinion you are mixing two different things: Initial Gitify/Composer setup AND Installing of MODX via Gitify. 

These are two totally different things imho. And inside the script you are also assuming that a windows-user has git installed (which is not default) without checking against it. You can check for commands inside bat-files: https://gist.github.com/sebastian-marinescu/079a5cab3ae987207116cfceea4c6dc4

So even though I like script-wizardry, this script is too opinionated and tries too much at once. I couldn't make use of it in my environment. Of course that doesn't mean others couldn't use it and see a benefit.

I personally installed composer via the downloadable exe and cloned Gitify in my workspace-folder. 
And this happened once and probably won't happen again soon. 

But what happens more often? 

- Using already installed/set up Gitify to install a MODX-instance locally into a project-folder
- Installing and setting up Composer/Gitify on the remote-server you are working on
- Using that installed Gitify to install a MODX-instance remotely into your webspace

That's why I have 3 different scripts _(bat/sh)_ for doing these tasks. As they are also opinionated from my personal workflow (especially the remote-scripts, as I'm working mostly on the same hoster) I don't think they would really be of use for another developer. 

As this is about _.bat_ I will share my **project-setup-backend.bat** in the name of Open-Source and for inspirational purposes to this discussion: 
https://gist.github.com/sebastian-marinescu/855d13783097b098b8d842bb449ed777

**Comment by Mark-H** [comment]
This would be a great addition to the wiki, could you update that instead @siteleas?

**Comment by siteleas** [comment]
Sorry folks, but I have decided after much time and investment in MODx to change to Silverstripe (did this about six months ago. Silverstripe is far easier to work with when it comes to modifying the CMS and it is way more easy to integrate third party tools and libraries as it relies heavily on composer. It's a far more natural fit for git integration and since MODx shifted focus to accessibility features, which by far most users will not use, the move forward to version 3 is dragging on. I don't have time to wait for MODx to modernize. I'm sure that MODx will get there one day, but by then I will have heavily invested in Silverstripe and will not likely be moving back. Thank you to all on the MODx team for all of your great work. MODx is a great system and switching off of it was a very hard decision, but we were simply to hindered all the time in our efforts toward building simple CMP solutions. On top of that MODx is great for developers that enjoy developing in a web based environment but for us, waiting for page reloads going from module, to plugin, to template, etc editor screens doesn't work any more. We have found that editing .php and .ss files for updating code creating classes, and working with objects, methods and properties, is the fix for us as we are developing robust database driven interfaces, that take too long to build in MODx. Once again, thank you all for your efforts. Keep up the good work. MODx is my close second and was my first pick for quite a while. It still would be if we didn't do so much high end development. We needed a robust and mature framework more than we needed a far better cms than Joomla, Wordpress, Drupal, etc.

**Comment by Mark-H** [comment]
Thanks @siteleas - good luck with your work with SilverStripe. There's definitely plenty of improvements to be done on the MODX end and I can totally understand it's not possible to sit around forever when you have clients to deal with! Hopefully improvements in 3.0 will come to the general public soon :)

**Comment by siteleas** [comment]
I would definitely say that major improvement that has been discussed in the MODx forums to remove re-invented wheels, make cli (command line interface) work using composer, github, and modernizing using updated frameworks and scripting languages will help a lot. It's a major rework, I know, but if MODx doesn't do it, then it will go from being a great CMS that was way ahead of its time to being way behind the pack. That would truly be a shame because MODx is a great CMS/development platform for certain scenarios. Due to the massive conflicts keeping two code bases would cause for our small team, we had to come to the arduous conclusion to break away from MODx and go with Silverstripe. It meant going back and starting from scratch, but the beautiful thing was that Silverstripe and MODx have similar approaches in some areas so that eased the transition. I hope that MODx has a strong future and that its developers bring the promised changes to reality. It would be a real shame to not see MODx on the forefront of web development years down the road.

One thing to note is that a serious motivator for us was a far simpler approach to building custom CMS interfaces (like MODx CMPs). In silverstripe, building "CMPs" is native to silverstripe. You don't have to learn a third party framework in order to build powerful and robust interfaces easily. The biggest drawback in certain areas is the documentation, which while it is strong in some aspects, lacks uniformity and completeness when it comes to the API. (I know this is really quite typical, but SilverStripe really could improve in this area and make things easier for those unfamiliar with the SilverStripe API).

Anyway, just thought I would follow up on your message and try to be an encouragement. As I said, MODx is great for certain kinds of projects, but we have moved out of that arena and are developing framework style projects working with source files rather than the build within the CMS approach the MODx CMS is geared for.

**Comment by siteleas** [comment]
As for updating the wiki, maybe someone could look at my contribution and add it. I am not using MODx anymore so I am not geared that way at this time. But Someone would likely benefit from the script or the steps that it follows.

**Pull request closed without merging**